packages/agent-utils/src/llm/Chat.ts

```diff
@@ -125,134 +92,36 @@ export class Chat {
     }
   }
 
-  public async fitToContextWindow(): Promise<void> {
-    if (this.tokens < this._maxContextTokens) {
-      return;
-    }
-
-    this._logger?.error(`! Max Tokens Exceeded (${this.tokens} / ${this._maxContextTokens})`);
-
-    // Start with "temporary" messages
-    await this._summarize("temporary");
-
-    if (this.tokens < this._maxContextTokens) {
-      return;
-    }
-
-    // Move onto "persistent" messages
-    await this._summarize("persistent");
-  }
-
-  public export(): ChatMessageLog {
-    return JSON.parse(JSON.stringify(this._msgLogs));
+  public cloneChatLogs(): ChatLogs {
+    return this._chatLogs.clone();
   }
 
-  public toString() {
+  public toString(): string {
     return JSON.stringify(this, null, 2);
   }
 
-  public toJSON() {
-    return this._msgLogs;
-  }
-
-  private _chunk(msg: ChatMessage): MessageLog {
-    const chunks: MessageLog = {
-      tokens: 0,
-      msgs: []
-    };
-    let content = msg.content || "";
-
-    while (content.length > 0) {
-      // Slice a chunk
-      const contentChunk = content.slice(0, this._chunkTokens);
-
-      // Append the chunk
-      chunks.tokens += this._tokenizer.encode(contentChunk).length;
-      chunks.msgs.push({
-        ...msg,
-        content: contentChunk
-      });
-
-      // Remove the chunk
-      content = content.slice(this._chunkTokens);
-    }
-
-    return chunks;
+  public toJSON(): unknown {
+    return this._chatLogs;
   }
 
-  private async _summarize(
-    msgType: MessageType
-  ): Promise<void> {
-    const msgLog = this._msgLogs[msgType];
-
-    const message = await this._summarizeMessages(msgLog.msgs);
-
-    if (!message) {
-      return;
+  public async fitToContextWindow(): Promise<void> {
```

Comment:
if this function is called it's because it is expecting `contextWindow` to be defined right? wouldn't it be better to gracefully handling this with an error if needed? :thinking: just thinking out loud

Reply:
I see what you mean, but my logic is this: "If a chat does not have a context window to fit itself to, then the `fitToContextWindow` method should succeed because it is not constrained"